app/assets/javascripts/discourse/lib/utilities.js.es6

```diff
@@ -274,7 +283,7 @@ export function authorizedImagesExtensions() {
 }
 
 export function authorizesAllExtensions() {
-  return Discourse.SiteSettings.authorized_extensions.indexOf("*") >= 0;
+  return rawExtensions().indexOf("*") >= 0;
```

Comment:
I don't think we need to concat the settings, I'll prefer if we just add another condition that checks against `siteSettings.authorized_staff_extensions` if the user is a staff.